__tests__/index.test.js

```diff
@@ -82,8 +82,8 @@ const makeContext = (body, headers) => {
     }
 }
 
-const mockIssueComment = (comment) => {
-    return makeContext({
+const testIssueComment = (comment, fn) => {
+    const context = makeContext(extend({
```

Comment:
`extend()` snuck in here, but only gets defined in the next commit.

Reply:
Right you are! I had not made sure that `git rebase -x "npm run test"` succeeds, sorry. Now force-pushed. The range-diff is:



* 1:  18607d5 !  1:  9e00a3e Support `npm run lint`

   ``````diff
   @@ package.json
         "jest": "^29.4.2"
    +  },
    +  "scripts": {
   -+    "lint": "eslint -c .eslintrc.json '**/*.js'",
   ++    "lint": "eslint -c .eslintrc.json '**/*.js'"
       }
     }
   ``````

* 2:  5afa4f8 =  2:  41e25fb npm run lint: handle Jest correctly
* 3:  e64cdf3 !  3:  ff496e9 jest --init

   ``````diff
   @@ jest.config.js (new)
    
     ## package.json ##
    @@
   +     "jest": "^29.4.2"
       },
       "scripts": {
   -     "lint": "eslint -c .eslintrc.json '**/*.js'",
   +-    "lint": "eslint -c .eslintrc.json '**/*.js'"
   ++    "lint": "eslint -c .eslintrc.json '**/*.js'",
    +    "test": "jest"
       }
     }
   ``````

* 4:  f5aa71e =  4:  5227939 jest.config: adjust indentation
* 5:  3d826ad =  5:  7736b7d Start the test suite
* 6:  1fa9080 =  6:  4d81bf4 Run the tests as part of the PR checks
* 7:  30373d4 =  7:  56ee27e tests: verify the remaining wehbook validations
* 8:  1dc54cf =  8:  0e36b66 tests: add a regression test for the `/hi` command
* 9:  b7b802b !  9:  b557b82 tests: show `context.log()` output upon failure

   ``````diff
   @@ __tests__/index.test.js: const makeContext = (body, headers) => {
    -const mockIssueComment = (comment) => {
    -    return makeContext({
    +const testIssueComment = (comment, fn) => {
   -+    const context = makeContext(extend({
   ++    const context = makeContext({
             action: 'created',
             comment: {
                 body: comment,
   @@ __tests__/index.test.js: const mockIssueComment = (comment) => {
         }, {
             'x-github-event': 'issue_comment'
         })
   ++
    +    test(`test ${comment}`, async () => {
    +        try {
    +            await fn(context)
   ``````

* 10:  a4b9663 ! 10:  d155922 testsIssueComment: prepare for additional body contents

   ``````diff
   @@ __tests__/index.test.js: const makeContext = (body, headers) => {
     }
     
    -const testIssueComment = (comment, fn) => {
   +-    const context = makeContext({
    +function extend (a, ...list) {
    +    for (const b of list) {
    +        for (const key of Object.keys(b)) {
   @@ __tests__/index.test.js: const makeContext = (body, headers) => {
    +    const repo = bodyExtra_?.repository?.name || 'git'
    +    const number = bodyExtra_?.issue?.number || 0
    +    const pullOrIssues = bodyExtra_?.issue?.pull_request ? 'pull' : 'issues'
   -     const context = makeContext(extend({
   ++    const context = makeContext(extend({
             action: 'created',
             comment: {
                 body: comment,
   @@ __tests__/index.test.js: const testIssueComment = (comment, fn) => {
    +    }, bodyExtra_ ? bodyExtra_ : {}), {
             'x-github-event': 'issue_comment'
         })
   -     test(`test ${comment}`, async () => {
   + 
   ``````

* 11:  e87b30a = 11:  5ae167a tests: also test `/open pr`
* 12:  af8529d = 12:  e9cbbec tests: also test `/deploy`
* 13:  c7bd4a1 = 13:  51550f3 Add a unit test for `component-updates`
* 14:  a140d03 = 14:  700b95b guess release notes: support GNU TLS tickets better
* 15:  f9c0bb5 = 15:  2ead6fe guess release notes: handle newer GCM tickets better
* 16:  afb96cc = 16:  2180943 tests: also test `/add release note`

Reply:
@rimrul does it look okay now?

Reply:
That looks good.